internal/host/host.go

```diff
@@ -1275,3 +1278,43 @@ func (m *Manager) GetHostByKubeKey(key types.NamespacedName) (*common.Host, erro
 func (m *Manager) UnRegisterHost(ctx context.Context, hostID, infraEnvID string) error {
 	return common.DeleteHostFromDB(m.db, hostID, infraEnvID)
 }
+
+func (m *Manager) GetKnownHostApprovedCounts(clusterID strfmt.UUID) (registered, approved int, err error) {
+	var hostCounts []struct {
+		Count    int
+		Approved bool
+	}
+	err = m.db.Table("hosts").Select("count(*) as count, approved").Group("approved").
+		Where("cluster_id = ? and status = ? and deleted_at is null", clusterID.String(), models.HostStatusKnown).Scan(&hostCounts).Error
+	if err != nil {
+		m.log.WithError(err).Errorf("Failed to get counts")
+		return
+	}
+	for _, h := range hostCounts {
+		registered += h.Count
+		if h.Approved {
+			approved += h.Count
+		}
+	}
+	return
+}
+
+func (m *Manager) HostWithCollectedLogsExists(clusterId strfmt.UUID) (bool, error) {
+	var hostExists []int
+	query := `SELECT 1 as host_exists 
+				WHERE EXISTS
```

Comment:
Maybe the choice is to start the string on the first line. Then it could be:
```
<tab>query := `
<tab><tab>SELECT 1 as host_exists 
<tab><tab>WHERE EXISTS (
<tab><tab><tab>SELECT 1 
<tab><tab><tab>FROM hosts 
<tab><tab><tab>WHERE
<tab><tab><tab><tab>cluster_id = ? AND 
<tab><tab><tab><tab>logs_collected_at <> ? AND 
<tab><tab><tab><tab>deleted_at IS NULL
<tab><tab>)`
```

But I've found that SQL formatting is very subjective so up to you if you want to do any more.
At least I would say that this line only needs to be indented at most one level more than the previous one.

Reply:
Also sorry for the `<tab>`s but I couldn't figure out a better way to get the spacing across in markdown :man_shrugging:

Reply:
Pushed another commit.  Not sure it is better than the one we started with.

Reply:
Pushed another version without the raw SQL